instrument_registry/lifespan.py

```diff
@@ -0,0 +1,40 @@
+import json
+import logging
+from collections.abc import AsyncGenerator
+from contextlib import asynccontextmanager
+from typing import Any, TextIO
+
+import yaml
+from fastapi import FastAPI
+from instrument_registry.core.config import PROJECT_NAME, VERSION, get_settings
+from instrument_registry.core.log import configure_logging
+
+cached_data: dict[str, Any] = {}
```

Comment:
This works, it might be worth looking into the @lru_cache annotation though, as that is 'by default' the easiest way to do caching in Python.

https://docs.python.org/3/library/functools.html#functools.lru_cache

Reply:
I am not sure if that would work in our context. We followed the documentation: https://fastapi.tiangolo.com/advanced/events/#use-case. It appears we want to have a global object `cached_data` so we can share this among all API endpoints. I am not sure how this would work if we add the LRU cache decorator.

Reply:
We understand you, but actually like the solution we implemented where we can access the cached_data anywhere in the code compared to calling cached functions. We took this ML code example as a reference https://fastapi.tiangolo.com/advanced/events/#use-case